### PR TITLE
Added missing export for the flutter engine group.

### PR DESCRIPTION
**Pull request by gaaclarke**
## Tests

I added the following tests:

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [x] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h [hunk 1]
```diff
@@ -19,6 +19,7 @@ NS_ASSUME_NONNULL_BEGIN
  * @warning This class is a work-in-progress and may change.
  * @see https://github.com/flutter/flutter/issues/72009
  */
+FLUTTER_EXPORT
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h:22]: Why do we add `__attribute__((visibility("default")))` to classes in our public headers?  I wouldn't have thought that would be necessary but I don't know the history of this, is `FLUTTER_EXPORT` defined to something else in some cases?

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h:22]: Nope, that's what it's defined as.  When we compile the library we have the default visibility set to private, it's a linker flag.  The reason for it is mostly around c++ code.  By being more restrictive on what symbols are public different optimizations can happen in the loader, on top of having a little bit more control over how people use your API.

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h:22]: > we have the default visibility set to private

Ah, gotcha.

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**